### PR TITLE
Aducm3029 adt7420

### PR DESCRIPTION
**Pull request by mirceacaprioru**
ADuCM3029_demo_adt7420: Add main application

    This patch adds the application source. It is modified to use the new BLE
    message format intended to work with IOTnode IOS app.

    This is just a port of the ADICUP3029 BSP adt7420 project with
    modifications made to the BLE message format.

    For information about the BLE message format check:
    https://wiki.analog.com/resources/eval/user-guides/eval-adicup3029/reference_designs/ble_packet

    Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

### CODE REVIEW

**Comment by mirceacaprioru** [comment]
Changelog v1->v2
- fix application crash based on degree symbol not recognized by the app ( °)

File: projects/ADuCM3029_demo_adt7420/include/adt7420_app.h [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+extern "C" int32_t adi_initpinmux(void);
+
+typedef union {
+	uint8_t u8Value[4];
+	float fValue;
+} floatUnion_t;
+
+#pragma pack(push)
+#pragma pack(1)
+typedef struct GenericReg {
```
Comment (adrimbarean) [inline projects/ADuCM3029_demo_adt7420/include/adt7420_app.h]: We should not use typedef for structs.

File: projects/ADuCM3029_demo_adt7420/src/adt7420_app.cpp [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+ * @param [in] argc : Number of arguments (unused)
+ *
+ * @param [in] argv : Arguments (unused)
+ *
+ */
+int main(int argc, char *argv[])
+{
+    ADT7420         adt7420;
+    Temperature     *pTemp = &adt7420;
+    SENSOR_RESULT   eSensorResult;
```
Comment (adrimbarean) [inline projects/ADuCM3029_demo_adt7420/src/adt7420_app.cpp]: This variable might be set but not used in Release mode.

**Comment by adrimbarean** [comment]
The Release version of the project does not include in the compiler path the include folder.

**Comment by adrimbarean** [comment]
The Release version of the project defines REDIRECT_OUTPUT_TO_UART in common.h:82:0 for DFP 3.2.0 and as a define flag for the compiler, which generates a warning.

**Comment by mirceacaprioru** [comment]
Changelog v2 -> v3

-fix release configuration
- remove brackets

**Comment by mirceacaprioru** [comment]
Will merge today if there are no objections.

**Pull request merged**